pkg/option/config.go

```diff
@@ -3354,8 +3371,23 @@ func InitConfig(programName, configName string) func() {
 			} else {
 				// replace deprecated fields with new fields
 				ReplaceDeprecatedFields(m)
-				err := MergeConfig(m)
-				if err != nil {
+				var confArgs []string
+				for key := range m {
+					value := fmt.Sprintf("%v", m[key])
+					// check whether the flag exists
+					if IsKnownFlag(key) {
+						if value != "" {
+							value = strings.Join([]string{"=", value}, "")
+							confArgs = append(confArgs, strings.Join([]string{"--", key, value}, ""))
+						}
+					}
+				}
+
+				if err := cmd.ParseFlags(confArgs); err != nil {
```

Comment:
The approach here converts each config key into its equivalent command line argument, and then parses the command line argument, which seems like a heavy way of validating the config. Is there any way to validate the config directly?

Reply:
> The approach here converts each config key into its equivalent command line argument, and then parses the command line argument, which seems like a heavy way of validating the config. Is there any way to validate the config directly?

In this case, need to develop a new parser. I've chosen the approach of reusing the code.